### ISSUE TITLE
Throwing error may inject format string?

### ISSUE THREAD

**Issue by Goncalerta** [specs-feup/clava#149]
`throw new Error("Isto é %s um bocado preocupante")`

![image](https://github.com/specs-feup/clava/assets/22060232/e1f9b453-fa9f-4dd9-a4e0-55eb1f096333)

My `%static` lifetimes aren't very happy with this 😛 

![image](https://github.com/specs-feup/clava/assets/22060232/f62f6aec-99b3-49fe-bc4f-a176e6c063a0)

**Comment by lm-sousa** [comment]
Where is that `throw` present?
I can't find it in the codebase.

**Comment by lm-sousa** [comment]
Perhaps an issue with [debug](https://github.com/debug-js/debug), or the way it is being used.

Either way, that sort of exceptions should be printed to `stderr`. I noticed that a while ago and need to correct it.

**Comment by lm-sousa** [comment]
Also, shouldn't you escape those characters anyway?

**Comment by Goncalerta** [comment]
> Also, shouldn't you escape the `%` character anyway in the `%static` case?

No, there is no reason to be escaping a `%` character. I'm currently doing so as a workaround because of this bug, but I thought I should report it anyway because these kinds of accidental format leaks may end up exposing vulnerabilities, so it's worth understanding what is happening here.

**Comment by lm-sousa** [comment]
What seems to be happening is that [debug](https://github.com/debug-js/debug) appends a `%s` to the end of the string to add the timestamp info but text gets captured by the first `%s` in the string.
This could be avoided if the string was sanitized by [debug](https://github.com/debug-js/debug) before appending their stuff.

![image](https://github.com/specs-feup/clava/assets/3999751/bf42f8b1-aa09-4f3d-828f-c597aac5c0ec)
